x-pack/plugins/lens/public/datasources/text_based/components/dimension_editor.tsx

```diff
@@ -24,14 +33,28 @@ export type TextBasedDimensionEditorProps =
 export function TextBasedDimensionEditor(props: TextBasedDimensionEditorProps) {
   const [allColumns, setAllColumns] = useState<FieldOptionCompatible[]>([]);
   const query = props.state.layers[props.layerId]?.query;
+  const { euiTheme } = useEuiTheme();
+  const {
+    isFullscreen,
+    columnId,
+    layerId,
+    state,
+    setState,
+    indexPatterns,
+    dateRange,
+    expressions,
+  } = props;
 
   useEffect(() => {
     // in case the columns are not in the cache, I refetch them
     async function fetchColumns() {
       if (query) {
         const table = await fetchFieldsFromESQL(
           { esql: `${query.esql} | limit 0` },
-          props.expressions
+          expressions,
+          { from: dateRange.fromDate, to: dateRange.toDate },
+          undefined,
+          indexPatterns?.length ? (indexPatterns[0] as unknown as DataView) : undefined
```

Comment:
`indexPatterns` is a `Record<string, IndexPattern>` why are you handling it as an array?  Do we use numeric keys for these `IndexPattersMap` object ?

Reply:
fetchFieldsFromESQL doesnt really need the dataview but just timefield name do i updated that as well so i dont need to cast here.

Reply:
addressed in https://github.com/elastic/kibana/pull/201540/commits/46eb97e931190afc842655c7afec8ef972b44d4c